### DISCUSSION TITLE
abnormal display of endpoints list

### DISCUSSION THREAD

**Discussion by BertOnline** [apache/skywalking#9650] [Q&A] [answered]
My service does not have these endpoints, but the list shows them. What causes this? The endpoints displayed end in vt_ 

![image](https://user-images.githubusercontent.com/17882970/191398397-238d27ed-5874-411d-911a-6190c5dbf8a6.png)

![image](https://user-images.githubusercontent.com/17882970/191398763-a2af6f29-8ad3-4f6a-9890-775f2bf13e7a.png)

**Comment by BertOnline** [comment]
The overview display is normal.

![image](https://user-images.githubusercontent.com/17882970/191399537-1e39582d-dc1f-4f89-9039-94499bcaccfd.png)

**Comment by wu-sheng** [comment]
SkyWalking doesn't manipulate your URI/endpoint. One possibility is, that a user or client accesses your system through these URI(s).

**Reply by BertOnline** [reply]
But, the overview display is normal. There is no such endpoint(/collection/vt_*) in service, and it shows that these endpoints have no requests

**Comment by BertOnline** [comment, marked as answer]
Thanks for answering the question. I see that the interface is normal on Prometheus, so I'm confused. I'll look for other reasons

![image](https://user-images.githubusercontent.com/17882970/191403051-4650f552-274a-40f4-afa9-3601bc14cf7e.png)

**Reply by wu-sheng** [reply]
What you show on Prometheus has nothing related to skywalking. Those are two different systems, use totally different ways to collect data. Check the original traces you collected.

**Reply by BertOnline** [reply]
@wu-sheng Thank you. The root cause of this problem has been found. Our team members have added a new interface. As shown in the figure below：
![image](https://user-images.githubusercontent.com/17882970/191433148-0bee35dc-a84e-4792-8dea-c70efb94f15f.png)

Due to the adoption of @PathVariable interface, the request is displayed as /collect/xxx,
Is there any way to combine and display @PathVariable interfaces, as shown in the following figure:
![image](https://user-images.githubusercontent.com/17882970/191433870-a5d8eb80-8adc-444c-bfe9-e5fceb379eeb.png)

Second, Whether the endpoint list can be displayed in pages.
![image](https://user-images.githubusercontent.com/17882970/191434514-53f7484f-d3f8-48dc-a8ea-d09124ce91c8.png)

Thanks again for your reply.

**Reply by wu-sheng** [reply]
> Is there any way to combine and display 

You could either enhance SpringMVC plugin to support stage annotation, I think we don't support that. But for path and controller annotations, we have supported.

Or you could use backend OpenAPI or regex based grouping mechanism in the backend. 

Generally, enhancing plugin could have better performance. Grouping at the server side costs more.

>Second, Whether the endpoint list can be displayed in pages.

No, number of endpoints are easy to be 10k+. Paging is a bad practice. Use the search for endpoint, it is the only reasonable way to show so many options.

**Reply by BertOnline** [reply]
@wu-sheng  We started to enhance SpringMVC plugin to support stage annotation. Thanks again.
